spark-cassandra-connector/src/main/scala/com/datastax/spark/connector/cql/CassandraConnectorConf.scala

```diff
@@ -60,26 +60,37 @@ object CassandraConnectorConf extends Logging {
     }
   }
 
-  def apply(conf: SparkConf): CassandraConnectorConf = {
-    val hostsStr = conf.get(CassandraConnectionHostProperty, InetAddress.getLocalHost.getHostAddress)
+  def processProperty(property: String, cluster: Option[String] = None): String = {
+    cluster match {
+      case Some(c) => property.replaceFirst("spark.cassandra", s"spark.$c.cassandra")
+      case None => property
+    }
+  }
```

Comment:
I think you can simplify

      def processProperty(property: String, cluster: Option[String] = None): String =  
          cluster.map(c => property.replaceFirst("spark.cassandra", s"spark.$c.cassandra")).getOrElse(property)